VirtoCommerce.Storefront/Domain/Common/SettingConverter.cs

```diff
@@ -1,22 +1,23 @@
+using System.Linq;
 using VirtoCommerce.Storefront.Model;
 using platformDto = VirtoCommerce.Storefront.AutoRestClients.PlatformModuleApi.Models;
 
 namespace VirtoCommerce.Storefront.Domain
 {
     public static class SettingConverter
     {
-        public static SettingEntry ToSettingEntry(this platformDto.Setting settingDto)
+        public static SettingEntry ToSettingEntry(this platformDto.ObjectSettingEntry settingDto)
         {
             var retVal = new SettingEntry();
-            retVal.DefaultValue = settingDto.DefaultValue;
-            retVal.Description = settingDto.Description;
-            retVal.IsArray = settingDto.IsArray ?? false;
+            retVal.DefaultValue = settingDto.DefaultValue?.ToString();
+            retVal.IsArray = false;
             retVal.Name = settingDto.Name;
-            retVal.Title = settingDto.Title;
-            retVal.Value = settingDto.Value;
+            retVal.Value = settingDto.Value?.ToString();
```

Comment:
You can't use Value.ToString() to cast object value. The ToString method of any object is supposed to return a string representation of that object and can produces  a different object string representation based on execution context thread culture for example. e.g  3.15.ToString() => "3,14" and after back conversation to decimal with invariant culture  will produce 314 decimal value. What exactly happens for tax settings.  https://github.com/VirtoCommerce/vc-storefront-core/blob/master/VirtoCommerce.Storefront/Domain/Stores/StoreService.cs#L118

Reply:
I propose to change the type to from `string` to `object` for these properties  `SettingEntry.Value`, `SettingEntry.DefaultValue`, `SettingEntry.AllowedValues` to prevent the object to string casting errors.